components/admin/users/AdminUsersSearchBar.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import _ from 'lodash'
+
+type SearchBarProps = {
+  searchOption: {
+    option: string
+    admin: string
+    searchTerm: string
+  }
+
+  setSearchOption: React.Dispatch<
+    React.SetStateAction<{
+      option: string
+      admin: string
+      searchTerm: string
+    }>
+  >
+}
+
+export const SearchBar: React.FC<SearchBarProps> = ({
+  searchOption,
+  setSearchOption
+}) => {
+  const run = _.debounce(setSearchOption, 500)
```

Comment:
you don't need debounce, because all the data is already there. you are just searching through existing data so this should be instant.

Reply:
Debounce is needed because when someone is typing in the searchbar, it becomes very laggy. I noticed that the users list was re-rendering way too often. This re-rendering issue was happening before I added in the option to `highlight` the searchTerm in the users list too.

Reply:
Could this be a problem similar to what we faced with the `ReviewCard` component? 

Using `memo` will not work in this instance because in the next PR, the component that displays a list of users is going to take in a `searchOption` property, whose value will change each time someone times in the search bar.

Reply:
thanks for verifying, I believe you and I wouldn't have guessed that. Could you add a comment in case someone else has the question?